### PR TITLE
Add ConfigUpdaterStep unit tests

### PR DESCRIPTION
**Pull request by mjrousos**
Starting to chip away at test debt. This adds tests for `ConfigUpdaterStep` and `ConfigUpdaterSubStep`.

### CODE REVIEW

File: tests/Microsoft.DotNet.UpgradeAssistant.TestHelpers/Microsoft.DotNet.UpgradeAssistant.TestHelpers.csproj [hunk 1]
```diff
@@ -10,4 +10,7 @@
     <ProjectReference Include="..\..\src\common\Microsoft.DotNet.UpgradeAssistant.Abstractions\Microsoft.DotNet.UpgradeAssistant.Abstractions.csproj" />
     <ProjectReference Include="..\..\src\components\Microsoft.DotNet.UpgradeAssistant.MSBuild\Microsoft.DotNet.UpgradeAssistant.MSBuild.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <Folder Include="Steps\" />
```
Comment (twsouthwick) [inline tests/Microsoft.DotNet.UpgradeAssistant.TestHelpers/Microsoft.DotNet.UpgradeAssistant.TestHelpers.csproj:14]: Is this needed?

File: tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterStepTests.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            var dependsOn = step.DependsOn;
+            var description = step.Description;
+            var id = step.Id;
+            var title = step.Title;
+            var subSteps = step.SubSteps;
+            var status = step.Status;
+            var done = step.IsDone;
+
+            // Assert
+            Assert.Equal(new[] { "Microsoft.DotNet.UpgradeAssistant.Steps.Solution.NextProjectStep" }, dependencyOf);
```
Comment (twsouthwick) [inline tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterStepTests.cs:46]: Should we keep a list of "well known steps" that we can just reference via a static class? These magic strings get difficult

File: tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterStepTests.cs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            Assert.Equal(
```
Comment (twsouthwick) [inline tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterStepTests.cs:47]: I've started using this for collections

```csharp
Assert.Collection(dependsOn.OrderBy(x => x),
  d => Assert.Equal("Microsoft....", d),
  d => Assert.Equal(....))
```

File: tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterStepTests.cs [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        }
+
+        [Theory]
+        [InlineData(new[] { "TestConfig.xml" }, new[] { "TestConfig.xml" }, 1, 1)] // Vanilla positive case
+        [InlineData(new[] { "TestConfig.xml", "DoesNotExist" }, new[] { "TestConfig.xml" }, 2, 2)] // Non-existent config file
+        [InlineData(new[] { "TestConfig.xml", "TestConfig.xml" }, new[] { "TestConfig.xml", "TestConfig.xml" }, 0, 1)] // Multiple config files
+        [InlineData(new[] { "TestConfig.xml" }, new[] { "TestConfig.xml" }, 1, 0)] // Vanilla positive case, complete
+        [InlineData(new[] { "TestConfig.xml", "DoesNotExist" }, new[] { "TestConfig.xml" }, 2, 0)] // Non-existent config file, complete
+        [InlineData(new[] { "TestConfig.xml" }, new[] { "TestConfig.xml" }, 0, 0)] // No config updaters
+        [InlineData(new string[0], new string[0], 2, 0)] // No config files
```
Comment (twsouthwick) [inline tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterStepTests.cs:91]: didn't know you could do arrays like this in attributes. Nice!

File: tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterSubStepTests.cs [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    public class ConfigUpdaterSubStepTests
+    {
+        [Fact]
+        public void CtorTests()
+        {
+            // Arrange
+            (_, var step) = GetMockContextAndStep(new[] { "a" }, true, new[] { new TestConfigUpdater(BuildBreakRisk.Medium, false) });
+
+            // Act
+            var risk = step.Risk;
```
Comment (twsouthwick) [inline tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterSubStepTests.cs:28]: Do we need all the local variables?

**Review by twsouthwick — APPROVED** [review]
Some small feedback but looks good

File: tests/Microsoft.DotNet.UpgradeAssistant.TestHelpers/Microsoft.DotNet.UpgradeAssistant.TestHelpers.csproj (hunk 1 above)
Comment (mjrousos) [reply tests/Microsoft.DotNet.UpgradeAssistant.TestHelpers/Microsoft.DotNet.UpgradeAssistant.TestHelpers.csproj:14]: Nope. Not sure why it's there. Removed.

File: tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterStepTests.cs (hunk 2 above)
Comment (mjrousos) [reply tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterStepTests.cs:46]: That's a good idea. I'll add this.

File: tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterSubStepTests.cs (hunk 5 above)
Comment (mjrousos) [reply tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration.Tests/ConfigUpdaterSubStepTests.cs:28]: No, my 'Act' section was just feeling a little lonely :) I think you're right that it's better without them. I'll inline them.

**Pull request merged**